File: packages/flare-core-apple-sdk/Sources/FlareCoreAppleSDK/Adapter/Module/DefaultUserApi.swift
import Foundation

/// User profile cache facade. Business pushes identity (name/avatar) here; reads
/// batch-join the cache to render current identity. Operation `user.upsert_profiles`
/// goes through the JSON contract invoke boundary (`flare_sdk_invoke_json`).
public final class DefaultUserApi: UserApiProtocol {
    private let bridge: any NativeBridgeProtocol

    public init(bridge: any NativeBridgeProtocol) {
        self.bridge = bridge
    }

    public func upsertUserProfiles(_ request: [String: AnySendable]) async throws -> Void {
        try await invokeVoid(bridge, descriptor: NativeCallMap.userUpsertProfiles, request: AnySendable(request))
    }
}

File: packages/flare-core-android-sdk/src/main/kotlin/com/flare/im/adapter/module/DefaultUserApi.kt
package com.flare.im.adapter.module

import com.flare.im.adapter.codec.*
import com.flare.im.api.user.UserApi
import com.flare.im.contract.NativeBridge
import com.flare.im.contract.NativeCallMap

/**
 * User profile cache facade. Business pushes identity (name/avatar) here; reads
 * batch-join the cache to render current identity. Operation `user.upsert_profiles`
 * goes through the JSON contract invoke boundary (`flare_sdk_invoke_json`).
 */
class DefaultUserApi(
    private val bridge: NativeBridge,
) : UserApi {

    override suspend fun upsertUserProfiles(request: Map<String, Any?>): Unit {
        invokeUnit(bridge, NativeCallMap.USER_UPSERT_PROFILES, request)
    }
}
